pinot-core/src/main/java/org/apache/pinot/core/query/selection/SelectionOperatorUtils.java

```diff
@@ -437,20 +389,26 @@ public static SelectionResults renderSelectionResultsWithoutOrdering(@Nonnull Li
    * @param dataSchema data schema.
    * @return column indices
    */
-  public static int[] getColumnIndices(@Nonnull List<String> selectionColumns, @Nonnull DataSchema dataSchema) {
+  public static int[] getColumnIndices(List<String> selectionColumns, DataSchema dataSchema) {
```

Comment:
Why do we need this if the order of columns as processed by the SelectionOperator is same as what user specified? I see you are using a function SelectionOperatorUtils.extractColumns and the comment seems to imply that broker relies on that ordering  to process so I am not sure why we need column indices

Reply:
For selection order-by queries, the column order in the DataSchema is not the same as what user specified, and that is the reason why we need to calculate the column indices in the DataSchema to get the correct selection columns.